lts.js

```diff
@@ -0,0 +1,79 @@
+const fs = require("fs/promises");
+const path = require("path");
+
+(async () => {
+  const LTS_SCHEDULE_URL =
+    "https://raw.githubusercontent.com/nodejs/Release/main/schedule.json";
+
+  const versions = await fetch(LTS_SCHEDULE_URL).then((r) => r.json());
+
+  const currentLTS = +Object.entries(versions)
+    .map(([version, metadata]) => ({
+      version: version.replace(/\D/g, ""),
+      lts: metadata.lts ?? false,
+    }))
+    .filter(({ lts }) => lts !== false)
+    .map((o) => ({ ...o, lts: new Date(o.lts).getTime() }))
+    .sort(({ lts: a }, { lts: b }) => a - b)
+    .filter(({ lts }) => lts < Date.now())
+    .pop().version;
+
+  const pkg = JSON.parse(await fs.readFile("./package.json"));
+  const engineMajor = +pkg.engines.node.split(".")[0].replace(/\D/g, "");
```

Comment:
It does, yeah, and that is very confusing. So I'm refactoring it so that the chain transforms the original list of Node versions into an array of past or present LTS version numbers, and then just popping off the top. The conversion to a numeric value will happen more explicitly within the chain.